Jibril/BinaryTree/week_May_16_22/Leaf_similar_trees.py

```diff
@@ -0,0 +1,25 @@
+from typing import Optional
+# Definition for a binary tree node.
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+class Solution:
+    def leafSimilar(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> bool:
```

Comment:
You didn't add iterative solution. If you are having trouble with that I can explain how to do iterative solution on Sunday.

Reply:
Added the iterative solution on this one.